include/pybind11/pybind11.h

```diff
@@ -1075,15 +1083,27 @@ class class_ : public detail::generic_type {
     }
 
     /// Uses return_value_policy::reference_internal by default
+    template <typename Getter, typename Setter, typename... Extra>
+    class_ &def_property(const char *name, const Getter &fget, const Setter &fset, const Extra& ...extra) {
+        return def_property(name,
+                cpp_function(fget, is_method<type>(*this)), cpp_function(fset, is_method<type>(*this)),
+                return_value_policy::reference_internal, extra...);
+    }
```

Comment:
I know; I don't like it that much.  The alternative is adding a move constructor that takes and ignores extra arguments (i.e. `template <typename... Extra> cpp_function(cpp_function &&f, const Extra &...)`), but I like that approach even less (mainly because it's too late to apply any of the Extra arguments).